subsys/net/lib/openthread/platform/radio.c

```diff
@@ -58,6 +66,42 @@ static struct ieee802154_radio_api *radio_api;
 static s8_t tx_power;
 static u16_t channel;
 
+static u16_t energy_detection_time;
+static u8_t  energy_detection_channel;
+static s8_t  energy_detected_value;
+
+ATOMIC_DEFINE(pending_events, pending_events_count);
+
+
+static inline bool is_pending_event_set(enum pending_events_t event)
+{
+	return atomic_test_bit(pending_events, event);
+}
+
+static void set_pending_event(enum pending_events_t event)
+{
+	atomic_set_bit(pending_events, event);
+	otSysEventSignalPending();
+}
+
+static void reset_pending_event(enum pending_events_t event)
+{
+	atomic_clear_bit(pending_events, event);
+}
+
+static inline void clear_pending_events(void)
+{
+	atomic_clear(pending_events);
+}
+
+void energy_detected(struct device *dev, s16_t max_ed)
+{
+	if (dev == radio_dev) {
+		energy_detected_value = max_ed;
+		set_pending_event(pending_events_detect_energy_done);
```

Comment:
Shouldn't we call `otSysEventSignalPending();` here?

Reply:
set_pending_event does it